daemon/daemon.go

```diff
@@ -745,9 +747,46 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 		return nil, err
 	}
 
+	if err := d.cleanupMounts(); err != nil {
+		return nil, err
+	}
+
 	return d, nil
 }
 
+// cleanupMounts umounts shm/mqueue mounts for old containers
+func (daemon *Daemon) cleanupMounts() error {
```

Comment:
Also, shouldn't this run as part of daemon shutdown as well? I'd prefer to not leave mounts around.